ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/EIP2124Test.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.manager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.eth.manager.ForkIdManager.ForkId;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class EIP2124Test {
+  private static final Logger LOG = LogManager.getLogger();
+
+  @Parameters(name = "{index}: {0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[][] {
+          // Mainnet test cases
+          {"Mainnet // First Homestead block", Network.MAINNET, 1150000L, "0x97c2c34c", 1920000L},
```

Comment:
This feels like a lot of duplication with ` org.hyperledger.besu.ethereum.eth.manager.ForkIdManagerTest` - What is being tested here that is not being tested there and can these two files be merged into one?

Reply:
They don't exactly test the same things. The new test class `EIP2124Test` is focused on testing the fork id sent in P2P status messages. It also performs the assertions on multiple combinations of chain heights. The `ForkIdManagerTest` is focus on testing the intermediate fork hashes generated to perform the peer check. I think they have different scopes of responsibilities. I would keep them both. Moreover, the new test class cases are aligned with some tests in the EIP specification and also the implementation of other clients.

Reply:
Simply put: `ForkIdManagerTest` tests the behaviour of `peerCheck` method whereas `EIP2124Test` tests the behaviour of `computeForkId` method used by `EthProtocolManager`.

Reply:
Are you ok if i keep both files ?

Reply:
There's too much duplication.  They are testing two things in the same java file, so they should at least be in the same file.  Also, the data sets (what the fork IDs of the well known networks are) should be normalized into one data set.

Reply:
It is not a well-documented feature in the genesis configuration that this can occur. Esp. with EthProtocolManager handshake as that has been a worry for us.

Reply:
I refactored both the implementation and the testing.

Reply:
And tested on almost all Ethereum networks, including consortium networks.